pkg/syncer/shared/helpers.go

```diff
@@ -54,3 +60,14 @@ func GetUpstreamResourceName(downstreamResourceGVR schema.GroupVersionResource,
 	}
 	return downstreamResourceName
 }
+
+// GetDNSID returns a unique ID for DNS object derived from the sync target name, its UID and workspace. It's
```

Comment:
the workspace is not the workspace of the SyncTarget here afaict, it's the user workspace containing the namespaces + workload resources to be synced.

Reply:
To be exhaustive, we should probably also include the workspace of the `SyncTarget` in this hash calculation.

Reply:
Here the workspace is the user workspace. The generated name is similar to the the generated name for the synctarget deployment, which does not contain the `SyncTarget` workspace. For consistency reason I wouldn't add it.